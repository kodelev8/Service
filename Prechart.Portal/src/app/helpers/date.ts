import { Injectable } from "@angular/core";
import * as moment from "moment";

@Injectable({ providedIn: 'root' })
export class DateHelper {

  formatDate(date: any) {
    return moment(date).local().format('yyyy-MM-DD');
  }

}
