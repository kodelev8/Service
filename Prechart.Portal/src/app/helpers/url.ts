import { Injectable } from "@angular/core";
import { environment } from "src/environments/environment";

@Injectable({ providedIn: 'root' })
export class UrlHelper {
  format(subUrl: string, path: string) {
    var getUrl = this.getUrl(subUrl);

    if (environment.production) {
      var host = environment.apiUrl.replace('[[subdomain]]', getUrl);
      return `${host}${path}`;
    }
    else {
      var host = environment.apiUrl.replace('[[port]]', getUrl);
      return `${host}${path}`;
    }
  }

  getUrl(domain: string) {
    var value: string = '';
    switch (domain) {
      case 'document': value = environment.document;
        break;
      case 'belastingen': value = environment.belastingen;
        break;
      case 'user': value = environment.user;
        break;
      case 'loonheffings': value = environment.loonheffings;
        break;
      case 'person': value = environment.person;
        break;
      case 'werkgever': value = environment.werkgever;
        break;
      default: value = '0000';
        break;
    }
    return value;
  }
}
