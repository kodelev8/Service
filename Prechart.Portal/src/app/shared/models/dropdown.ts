export class Dropdown {
  public value: number;
  public label: string;
}
