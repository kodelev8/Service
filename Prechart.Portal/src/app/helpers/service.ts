import { HttpClient, HttpHeaders } from "@angular/common/http";
import { Injectable } from "@angular/core";
import { AccountService } from "../shared/account/services/account.service";
import { UrlHelper } from "./url";
import {User} from "../shared/account/models/user";

@Injectable({ providedIn: 'root' })
export class ServiceHelper {
  headers: HttpHeaders;
  constructor(
    private http: HttpClient,
    private accountService: AccountService,
    private urlHelper: UrlHelper)
  {
    if (this.accountService.userValue) {
      var token = this.accountService.userValue;

      this.headers = new HttpHeaders({
        'Authorization': `Bearer ${token}`,
      });
    }
  }

  get(port: string, path: string) {
    var url = this.urlHelper.format(port, path);
    return this.http.get<any>(url, { headers: this.headers });
  }

  post(port: string, path: string, object: any) {
    var url = this.urlHelper.format(port, path);
    return this.http.post<any>(url, object, { headers: this.headers });
  }

  validateUser(port: string, path: string, object: any, options: any) {
    var url = this.urlHelper.format(port, path);
    return this.http.post<any>(url, object, options);
  }
}
