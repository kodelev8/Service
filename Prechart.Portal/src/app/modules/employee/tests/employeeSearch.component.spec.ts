import { ComponentFixture, TestBed } from '@angular/core/testing';
import { HttpClientTestingModule } from '@angular/common/http/testing';
import { FormsModule } from '@angular/forms';
import { By } from '@angular/platform-browser';
import { HttpClientModule } from '@angular/common/http';
import { RouterTestingModule } from '@angular/router/testing';

import { EmployeeSearchComponent } from 'src/app/modules/employee/search/employeeSearch.component';
import { ServiceHelper } from '../../../helpers/service';
import { AccountService } from '../../../shared/account/services/account.service';
import { SidebarModule } from 'primeng/sidebar';
import { EmployeeDetailsComponent } from '../search/details/employeeDetails.component';
import { DividerModule } from 'primeng/divider';
import { Panel, PanelModule } from 'primeng/panel';
import { DropdownModule } from 'primeng/dropdown';
import { BrowserAnimationsModule } from '@angular/platform-browser/animations';

describe('EmployeeSearchComponent', () => {
  let fixture: ComponentFixture<EmployeeSearchComponent>;
  let component: EmployeeSearchComponent;
  let element: any;

  const getMockFn = jest.fn();
  let service: ServiceHelper;

  beforeEach(async () => {
    await TestBed.configureTestingModule({
      imports: [
        HttpClientTestingModule,
        HttpClientModule,
        RouterTestingModule,
        BrowserAnimationsModule,
        FormsModule,
        SidebarModule,
        DividerModule,
        PanelModule,
        DropdownModule,
      ],
      declarations: [
        EmployeeSearchComponent,
        EmployeeDetailsComponent,
      ],
      providers: [AccountService],
    }).compileComponents();
  });

  beforeEach(() => {
    fixture = TestBed.createComponent(EmployeeSearchComponent);
    component = fixture.componentInstance;
    element = fixture.nativeElement;
    fixture.detectChanges();
    service = TestBed.inject(ServiceHelper);

    service.get = getMockFn;
  });

  it('should create the Employee Search page', () => {
    expect(component).toBeTruthy();
  });

  it('should render the Employee Search page correctly', () => {
    expect(element).toMatchSnapshot();
  });

});
