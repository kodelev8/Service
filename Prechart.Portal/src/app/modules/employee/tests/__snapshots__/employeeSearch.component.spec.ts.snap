// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`EmployeeSearchComponent should render the Employee Search page correctly 1`] = `
<div
  id="root1"
>
  <div
    class="div-center p-2"
  >
    <h3>
      Onboarding Werknemers
    </h3>
    <p-dropdown
      class="p-element p-inputwrapper ng-untouched ng-pristine ng-valid"
      optionlabel="label"
      placeholder="Selecteer een werkgever"
    >
      <div
        class="p-dropdown p-component"
      >
        <div
          class="p-hidden-accessible"
        >
          <input
            aria-expanded="false"
            aria-haspopup="listbox"
            placeholder="Selecteer een werkgever"
            readonly=""
            role="combobox"
            type="text"
          />
        </div>
        
        <span
          class="p-dropdown-label p-inputtext p-placeholder"
        >
          Selecteer een werkgever
        </span>
        
        
        
        <div
          aria-haspopup="listbox"
          aria-label="dropdown trigger"
          class="p-dropdown-trigger"
          role="button"
        >
          <span
            class="p-dropdown-trigger-icon pi pi-chevron-down"
          />
        </div>
        
      </div>
    </p-dropdown>
    
    
    
    
  </div>
  <p-sidebar
    class="p-element"
    position="right"
    styleclass="p-sidebar-md"
  >
    
  </p-sidebar>
</div>
`;
